### DISCUSSION TITLE
Document reorganization not working (conditional compilation without #if)

### DISCUSSION THREAD

**Discussion by tmx7777** [codecadwallader/codemaid#821] [Q&A]
I'm working on a multi target project (incl. .NET 3.5, 4.8, 5.0). I successfully got rid of `#if` by handling conditional compilation within the `Directory.Build.targets` file. So files are named like `MyClass.Net35.cs` or `MyClass.Net48.cs`, etc. but they do not contain any `#if`  directives anymore. Nonetheless, CodeMaid does somehow think they contain `#if` and does not reorganize the files (i. e. sorting members, etc.) 

Are there any workarounds or settings to override this behaviour?

Thanks in advance for any help.

**Comment by codecadwallader** [comment]
Yes, there is a setting at CodeMaid->Options->Reorganizing->General->Perform reorganization when preprocessor conditional are present which you can change.  It is set to ask (dialog prompt) by default but that may have been changed.

We do check for some other pre-processor directives as well (e.g. `#else` and `#pragma`) so it may be another preprocessor statement that is triggering the logic.

![image](https://user-images.githubusercontent.com/3823772/126080067-8d7c94d3-9402-4d7d-8382-47af7c9cd2e8.png)

Hope it helps!
